app/src/main/java/app/security/RequiresPermissionsInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package app.security;
+
+import io.micronaut.aop.InterceptorBean;
+import io.micronaut.aop.MethodInterceptor;
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.core.type.MutableArgumentValue;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.exceptions.HttpStatusException;
+import jakarta.inject.Singleton;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Singleton
+@InterceptorBean(RequiresPermissions.class)
```

Comment:
This is cool, I hadn't seen it used before.

Reply:
Indeed! It's making use of https://docs.micronaut.io/3.10.1/guide/index.html#aroundAdvice

Reply:
Thanks for the link :smiley: